### ISSUE TITLE
Usage instructions

### ISSUE THREAD

**Issue by hellocatfood** [surfacepatterns/synthclone#27]
Are there any usage instructions or documentation on how to use synthclone?

**Comment by surfacepatterns** [comment]
Not really.  There really should be usage instructions.  There used to be instructions, but, when the `git` repo was imported from `googlecode.com` to `github.com`, the instructions that I created were lost.  I should have paid more attention to what was being imported.  I hope to remedy that problem in the future.

The basic steps are:
* Enable the participants that you'll need to clone the instrument you want to clone.  At a minimum, this means enabling a participant to capture samples (`JACK` or `PortMedia`, for example) and a participant that allows you to create a target corresponding to the export file format you want to create (e.g. if you're creating a Hydrogen drum kit, enable the `Hydrogen` participant).
* Add the sampler corresponding to the participant you enabled to capture samples.  You might need to configure things a bit.
* Add the target corresponding to the participant you enabled to export data.  You might need to configure things a bit.
* Setup zones for each of the notes you want to capture.  The `Zone Generator` participant can be helpful in setting up zones automatically over a range of notes, velocities, etc.
* Capture dry samples for all of the zones.
* Playback the dry samples, and recapture dry samples you're not happy with.
* If you want to apply effects, then enable participants for the effects you want to apply.  The `Trimmer` is almost always useful for getting rid of silence in the samples you capture.
* After choosing effects, apply the effects to the dry samples to yield wet samples.
* Playback the wet samples, and, if you're unhappy with the way the wet samples came out, adjust effect settings and reapply effects.
* Once you're happy with what you have, build your target(s).
* Test your targets and make sure they work.

There used to be a few videos floating around that folks did on how to use `synthclone`, but I'm having trouble finding them now.

If you have specific questions that aren't answered by the workflow above, feel free to ask, and I'll answer when I can.

**Comment by hellocatfood** [comment]
Thanks for that. I saw on a [few links](https://linuxmusicians.com/viewtopic.php?t=13687) that there used to be this video tutorial https://vimeo.com/73053302 but it appears to have disappeared now

**Comment by surfacepatterns** [comment]
> Thanks for that. I saw on a [few links](https://linuxmusicians.com/viewtopic.php?t=13687) that there used to be this video tutorial https://vimeo.com/73053302 but it appears to have disappeared now

I assume that's because '`synthclone` wasn't maintained for a significant amount of time.  Perhaps, when the next release is made, folks will start contributing videos, tutorials, etc. again. :)

**Comment by Gerbmouse** [comment]
Hi Surfacepatterns,
I have the feeling you are still on your way to rewrite Synthclone - that would be wonderful for it is really needed within the Linux audio ecosystem! I haven't found anything comparable.
I'd also like to ask a question about its usage. I'd like to sample the sound of a midi capable synth. I have setup synthclone like above described. Now I only find the possibility to launch the sampling of every key and velocity by pressing "Get Sample" individually by myself. That's very cumbersome. I have to sit at the screen and press "Get Sample" each e.g. 6 seconds again and again for more than 1000 times.
Do I miss the possibility that synthclone can do it automated and if yes, how can I accomplish it?

**Comment by surfacepatterns** [comment]
> I have the feeling you are still on your way to rewrite Synthclone - that would be wonderful for it is really needed within the Linux audio ecosystem! I haven't found anything comparable.

I've been spending chunks of my spare time working on the rewrite.  If you're curious, [this branch](https://github.com/surfacepatterns/synthclone/tree/surfacepatterns/rewrite) contains the current rewrite progress.  Everything compiles, and all the tests pass, but the application isn't usable yet.

> I'd also like to ask a question about its usage. I'd like to sample the sound of a midi capable synth. I have setup synthclone like above described. Now I only find the possibility to launch the sampling of every key and velocity by pressing "Get Sample" individually by myself. That's very cumbersome. I have to sit at the screen and press "Get Sample" each e.g. 6 seconds again and again for more than 1000 times. Do I miss the possibility that synthclone can do it automated and if yes, how can I accomplish it?

I haven't played with that version of `synthclone` in a **very** long time but, IIRC, you can select all of the zones and choose to get all the samples at the same time.  Unfortunately, I don't remember the details.

I appreciate you bringing this up though because I want to make sure that I get the workflow right for the rewrite.  If you have ideas on how this should ideally work, I'd love to hear them.

**Comment by Gerbmouse** [comment]
Whow, the fastest answer in the west, thanks a lot ;-) I surely will give the new branch a shot.

Concerning my original problem you gave me the correct hint. Selecting more than one line and pressing Ctrl-G starts sampling all highlighted lines one after the other (for sure not at the same time ;-). That solves my problem. I thank you very much!

**Comment by surfacepatterns** [comment]
> Whow, the fastest answer in the west, thanks a lot ;-)

Sure, NP. :D

> I surely will give the new branch a shot.

Nothing works there yet, really, save for tests.  There's barely a skeleton of a UI.  The plugin libraries are pretty solid, I guess. :)  However, if you're a developer (in particular, a C++ developer), then it might be of some interest.

> Concerning my original problem you gave me the correct hint. Selecting more than one line and pressing Ctrl-G starts sampling all highlighted lines one after the other (for sure not at the same time ;-). That solves my problem. I thank you very much!

I'm really glad to hear that. :)
